src/hooks/useLoadClientScript.ts

```diff
@@ -19,15 +20,47 @@ export function useLoadClientScript(props: NostoScriptProps) {
       }
       setClientScriptLoaded(true)
     }
-
-    // Create script element
-    function createScriptElement(urlPartial: string) {
+  
+    // Create and append script element
+    function injectScriptElement(urlPartial: string, extraAttributes?: Record<string, string>) {
```

Comment:
consider adding a default value for extraAttributes

Reply:
Sure, but the optionality needs to be dropped

Reply:
parameter with default value if effectively optional from usage POV